test/src/org/omegat/core/dictionaries/LingvoDSLTest.java

```diff
@@ -0,0 +1,78 @@
+/**************************************************************************
+ OmegaT - Computer Assisted Translation (CAT) tool
+          with fuzzy matching, translation memory, keyword search,
+          glossaries, and translation leveraging into updated projects.
+
+ Copyright (C) 2015 Hiroshi Miura
+               Home page: http://www.omegat.org/
+               Support center: http://groups.yahoo.com/group/OmegaT/
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **************************************************************************/
+
+package org.omegat.core.dictionaries;
+
+import java.io.File;
+import java.util.Map;
+
+import org.junit.Test;
+
+import org.omegat.core.TestCore;
+
+/**
+ * Dictionary test
+ *
+ * @author Hiroshi Miura
+ */
+public class LingvoDSLTest extends TestCore {
+
+    @Test
+    public void testReadFileDict() throws Exception {
+        LingvoDSL l = new LingvoDSL(new File("test/data/dicts/test.dsl"));
+        Map<String, Object> map = l.readHeader();
+        assertEquals(6, map.size());
+        
+        String word = "space";
+        String expect = "Only a single white space on first character\n";
+        Object data = map.get(word);
+        assertNotNull(data);
+        String result = l.readArticle(word, data);
+        assertEquals(expect, result);
+    }
+
+    @Test
+    public void testReadArticle1() throws Exception {
+        LingvoDSL l = new LingvoDSL(new File("test/data/dicts/test.dsl"));
+        Map<String, Object> map = l.readHeader();
+        String word = "tab";
+        String expect = "Translation line also can have a single TAB char\n";
+        Object data = map.get(word);
+        String result = l.readArticle(word, data);
+        assertEquals(expect, result);
+    }
+
+    @Test
+    public void testReadArticleRussian() throws Exception {
+        LingvoDSL l = new LingvoDSL(new File("test/data/dicts/test.dsl"));
+        Map<String, Object> map = l.readHeader();
+        String word = "tool";
+        Object data = map.get(word);
+        assertNotNull(data);
+        String result = l.readArticle(word, data);
+        assertEquals("\u0441\u0442\u0430\u043d\u043e\u043a\n", result); //станок
```

Comment:
As a general rule, we don't allow non-ASCII characters in the source, as they will turn into mojibake in some environments.